### PR TITLE
Refactor for e-h v1.0.0-alpha

### PR DESCRIPTION
**Pull request by ryankurte**
Blocked on e-h v1.0.0 release (https://github.com/rust-embedded/embedded-hal/issues/177)

cc. @therealprof

### CODE REVIEW

File: src/timer.rs [hunk 1]
```diff
@@ -132,8 +131,9 @@ impl Timer<SYST> {
 
 impl CountDown for Timer<SYST> {
     type Time = Hertz;
+    type Error = nb::Error<Error>;
```
Comment (eldruin) [inline src/timer.rs:134]: Does this need to be `nb::Error<Error>`? I think could just be `Error`.

Comment (ryankurte) [reply src/timer.rs:134]: the `try_wait` function below returns `nb::Error::WouldBlock` so, without changing this, yes?

Comment (ryankurte) [reply src/timer.rs:134]: wait you’re right i think i have this backwards, will try fixing it tomorrow. thanks for the comment!

Comment (ryankurte) [reply src/timer.rs]: got it, thanks for the review!

**Comment by maxekman** [comment]
Hi! How does this compare to my v1.0.0-alpha.1 branch #144? Can we merge/split the effort? I mainly want to be compatible with 1.0.x for the development of #145.

**Comment by ryankurte** [comment]
i somehow missed your PR when i made this one, they should be functionally equivalent. i haven't diffed em to check but, happy to close this in favour of yours and add the delta if there is anything missing.

**Comment by maxekman** [comment]
Sure, if that works for you @ryankurte. I'll see if I can finish it soon.

Do you think it could be useful to have a v1.0.0-dev branch for this repo so we can both merge any changes to it?

**Comment by maxekman** [comment]
There is now some new info about how to support both 0.2.x and 1.0.0-alpha: https://github.com/rust-embedded/embedded-hal#adding-support-for-an-embedded-hal--alpha-version-in-a-hal-implementation

**Comment by ryankurte** [comment]
while it's possible to do this it's also pretty easy to build adaptors between versions, and to update things, so i definitely would not think it an _expectation_ that both are supported.

**Comment by burrbull** [comment]
Closed in favor of #388

**Pull request closed without merging**